File: funciones.cpp
#include "varibales.h"
#include "iostream"
#include "string"

//declaracion de funciones
void agregarTarea();
void CompletarTarea();
void elminarTarea();
void actualizarTarea();
void verLista();

// programacion de funciones
int id = 0;
void agregarTarea(){
    cout << endl;
    tareas[id].id = id;
    cout << "Id de tarea: " << tareas[id].id+1<< endl;
    cout << "Que tarea deseas agregar: ";
    cin >> tareas[id].tarea;
    tareas[id].estado = "Pendiente";
    id++;
}
//esto encuentra la posicion en el arreglo mendiante un bucle for
int buscarId(int opcion){

    for (int i = 0; i < id; i++)
    {
        if (tareas[i].id == opcion)
        {
            return i;
        }
   }
     cout <<"No se encontro el id" << endl;
     return -1;
}

void verLista(){
for (int i = 0; i < id; i++)
{
    cout << "=============================" <<endl;
    cout <<"Id: " << tareas[i].id+1 << endl;
    cout << "Tarea: " <<  tareas[i].tarea<< endl;
    cout << "Estado " << tareas[i].estado<< endl;
}
}

void CompletarTarea(){
int opcion;
int res;
cout << endl;
cout << "Escribe el id de la tarea que has completado: " ;
cin >> opcion;
 res = buscarId(opcion-1);
 if (res == -1)
 {
    return;
 }
 tareas[res].estado = "Completado";
 cout << "Tarea " << tareas[res].tarea <<" marcada como completada. " << endl;
 }

void actualizarTarea(){
int opcion;
int res;
cout << endl;
cout << "Escribe el id de la tarea deaseas editar: ";
cin >> opcion;
 res = buscarId(opcion-1);
 if (res == -1)
 {
    return;
 }
 cout << "Edita el nombre de la tarea: ";
 cin >> tareas[res].tarea;
 cout << "Tarea actualizada a " << tareas[res].tarea << endl;
}


void eliminarTarea() {
    int opcion;
    cout << endl;
    cout << "Ingrese el ID de la tarea a eliminar: ";
    cin >> opcion;
    int index = buscarId(opcion-1);
    if (index != -1) {
        for (int i = index; i < id - 1; i++) {
            tareas[i] = tareas[i + 1];
            tareas[i].id--;
        }
        id--;
        cout << "Tarea eliminada exitosamente." << endl;
    } else {
        cout << "No se encontro la tarea con el ID proporcionado." << endl;
    }
}

File: toDoPyton/funciones.py
class Task:
    def __init__(self, task_id, name, description, importance):
        self.task_id = task_id
        self.name = name
        self.description = description
        self.importance = importance
        self.status = "Pendiente"

    def __str__(self):
        return (f"ID: {self.task_id}\n"
                f"Nombre: {self.name}\n"
                f"Descripción: {self.description}\n"
                f"Importancia: {self.importance}\n"
                f"Estado: {self.status}\n")

class TaskManager:
    def __init__(self):
        self.tasks = []
        self.next_id = 0

    def agregarTarea(self):
        name = input("Ingresa el nombre de la tarea: ")
        description = input("Ingresa la descripción de la tarea: ")
        importance = input("Ingresa la importancia de la tarea del 1 al 10: ")
        task = Task(self.next_id, name, description, importance)
        self.tasks.append(task)
        self.next_id += 1
        print("Tarea agregada.")
       
    def mostarTareas(self):
        for task in self.tasks:
            print("================================")
            print(task)
    
    def eliminarTareas(self):
        task_id = int(input("Escribe el ID de la tarea que deseas eliminar: "))
        for task in self.tasks:
            if task.task_id == task_id:
                self.tasks.remove(task)
                print("Tarea Eliminada")
                return
        print("No se encontró la tarea")
    def ActualizarTareas(self):
        task_id = int(input("Escribe el ID de la tarea que deseas actualizar: "))
        for task in self.tasks:
            if task.task_id == task_id:
                
                name = input("Escribe el nuevo nombre de la tarea: ")
                description = input("Escribe la nueva descripción de la tarea: ")
                importance = int(input("Escribe la nueva importancia de la tarea del 1 al 10: "))
                task.name = name
                task.description = description
                task.importance = importance
                print("Tarea actualizada.")
                return
        print("No se encontro la tarea.")
    def CompletarTarea(self):
        task_id = int(input("Escribe el ID de la tarea que deseas actualizar: "))
        for task in self.tasks:
            if task.task_id == task_id:
                task.status = "Completado"
                print("Tarea marcada como completada")
                return
        print("Tarea no encontrada")

                 
task_manager = TaskManager()
    


